lib/src/visitor/serialize.dart

```diff
@@ -1138,7 +1138,13 @@ class _SerializeVisitor
 
     var previousSpan = previous.span;
     if (previous is CssParentNode && previous.children.contains(node)) {
-      var endOffset = math.max(0, previousSpan.text.indexOf("{"));
+      // Walk back from just before the current node starts looking for the
+      // parent's left brace (to open the child block).  This is safer than
+      // a simple forward search of the previousSpan.text as that might contain
+      // other left braces.
+      var endOffset =
+          previousSpan.text.lastIndexOf("{", node.span.start.offset - 1);
```

Comment:
I don't think this logic is quite right... `node.span.start.offset` is an index into the entire source file, while `previousSpan.text`, just covers the span itself. I think you'll need to access [`node.span.file.getText()`](https://pub.dev/documentation/source_span/latest/source_span/SourceFile/getText.html).

Reply:
ah, I thought it was an offset relative to the beginning of the region covered by the span.  But I see now the FileLocation APIs are all absolute (which makes sense).  

I made the test case a little richer to try to catch the issue and then updated isTrailingComment to convert the absolute offset to relative to previousSpan's start offset: https://github.com/nickbehrens/dart-sass/commit/35f51df40734af3653e8f049d1915cbf61a32704

(I didn't see a need to use node.span.file.getText() but maybe you had something else in mind?)